src/ucp/core/ucp_ep.c

```diff
@@ -2504,6 +2504,13 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
         goto err;
     }
 
+    if ((key->err_mode == UCP_ERR_HANDLING_MODE_PEER) &&
+        (context->rcache == NULL)) {
+        ucs_warn("worker %p: memory invalidation is requested but not supported"
+                 " since rcache isn't initialized. Invalidation case will cause"
+                 " runtime failure.", worker);
```

Comment:
i'd also return error here (then can use `ucs_error` instead of `warn`)

Reply:
maybe we print this message in case of that runtime failure and not during initialization?

Reply:
ok